diff --git a/tools_webrtc/android/release_aar.py b/tools_webrtc/android/release_aar.py
index 1ac21b4280..68e37a131a 100644
--- a/tools_webrtc/android/release_aar.py
+++ b/tools_webrtc/android/release_aar.py
@@ -38,14 +38,14 @@ from android.build_aar import BuildAar
 
 
 ARCHS = ['armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64']
-MAVEN_REPOSITORY = 'https://google.bintray.com/webrtc'
+MAVEN_REPOSITORY = 'https://dl.bintray.com/crosstalk/maven'
 API = 'https://api.bintray.com'
-PACKAGE_PATH = 'google/webrtc/google-webrtc'
+PACKAGE_PATH = 'crosstalk/maven/crosstalk-webrtc'
 CONTENT_API = API + '/content/' + PACKAGE_PATH
 PACKAGES_API = API + '/packages/' + PACKAGE_PATH
-GROUP_ID = 'org/webrtc'
-ARTIFACT_ID = 'google-webrtc'
-COMMIT_POSITION_REGEX = r'^Cr-Commit-Position: refs/heads/master@{#(\d+)}$'
+GROUP_ID = 'io/crosstalk'
+ARTIFACT_ID = 'crosstalk-webrtc'
+COMMIT_POSITION_REGEX = r'^Cr-Commit-Position: refs/branch-heads/(\d+)@{#(\d+)}$'
 API_TIMEOUT_SECONDS = 10.0
 UPLOAD_TRIES = 3
 # The sleep time is increased exponentially.
@@ -92,7 +92,7 @@ def _GetCommitPos():
   if not commit_pos_match:
     raise Exception('Commit position not found in the commit message: %s'
                       % commit_message)
-  return commit_pos_match.group(1)
+  return commit_pos_match.group(1) + '.' + commit_pos_match.group(2)
 
 
 def _UploadFile(user, password, filename, version, target_file):
@@ -231,7 +231,7 @@ def _DeleteUnpublishedVersion(user, password, version):
 
 
 def ReleaseAar(use_goma, skip_tests, publish, build_dir):
-  version = '1.0.' + _GetCommitPos()
+  version = '84.' + _GetCommitPos()
   commit = _GetCommitHash()
   logging.info('Releasing AAR version %s with hash %s', version, commit)
 
diff --git a/tools_webrtc/android/templates/pom.jinja b/tools_webrtc/android/templates/pom.jinja
index 90ca51ca9c..9c9324ed91 100644
--- a/tools_webrtc/android/templates/pom.jinja
+++ b/tools_webrtc/android/templates/pom.jinja
@@ -2,8 +2,8 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <groupId>org.webrtc</groupId>
-  <artifactId>google-webrtc</artifactId>
+  <groupId>io.crosstalk</groupId>
+  <artifactId>crosstalk-webrtc</artifactId>
   <version>{{ version }}</version>
   <packaging>aar</packaging>
 
