diff --git a/rtc_base/physical_socket_server.cc b/rtc_base/physical_socket_server.cc
index 0cbcb8130c..b8ef8ea19d 100644
--- a/rtc_base/physical_socket_server.cc
+++ b/rtc_base/physical_socket_server.cc
@@ -207,7 +207,9 @@ int PhysicalSocket::Bind(const SocketAddress& bind_addr) {
         // and not try to use the socket. Otherwise, we may end up sending
         // packets with an invalid source address.
         // See: https://bugs.chromium.org/p/webrtc/issues/detail?id=7026
-        return -1;
+        if (!bind_addr.IsPrivateIP()) {
+          return -1;
+        }
       }
     }
   }
